programs/marginfi/src/state/marginfi_account.rs

```diff
@@ -246,7 +246,18 @@ impl<'a, 'b> BankAccountWithPriceFeed<'a, 'b> {
     ) -> MarginfiResult<I80F48> {
         match bank.config.risk_tier {
             RiskTier::Collateral => {
-                let price_feed = self.try_get_price_feed()?;
+                let price_feed = self.try_get_price_feed();
+
+                if matches!(
+                    (&price_feed, requirement_type),
+                    (&Err(PriceFeedError::StaleOracle), RequirementType::Initial)
+                ) {
+                    debug!("Skipping stale oracle");
+                    return Ok(I80F48::ZERO);
```

Comment:
I feel like instead of returning 0 we could use the std::Result<T,E> and return an error in this case and handle it one layer up. A stale oracle feels more like an error

Reply:
The purpose of this PR is this very behavior.